tools/block-generator/runner/run.go

```diff
@@ -90,7 +90,7 @@ func (r *Args) run() error {
 	}
 
 	// This middleware allows us to lock the block endpoint
-	var freezeMutex sync.Mutex
+	var freezeMutex deadlock.Mutex
```

Comment:
Curious to understand the benefits of `deadlock`'s **Mutex** over `sync`'s in this context.

Reply:
I believe it's just for the deadlock detection feature, you can see it configured here: https://github.com/algorand/go-algorand/blob/master/cmd/algod/main.go#L269